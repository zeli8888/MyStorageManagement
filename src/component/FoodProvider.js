import React, { Component, useState, useEffect } from 'react';
import IngredientService from '../service/IngredientService'
import DishService from '../service/DishService'

export const FoodContext = React.createContext();  //exporting context object

const FoodProvider = ({ children }) => {
    const [ingredients, setIngredients] = useState([]);
    const [dishes, setDishes] = useState([]);

    useEffect(() => {
        IngredientService.getAllIngredients().then(response => { setIngredients(response.data) })
        DishService.getAllDishes().then(response => { setDishes(response.data) })
    }, []);

    return (
        <FoodContext.Provider value={
            {
                ingredients: ingredients,
                setIngredients: setIngredients,
                dishes: dishes,
                setDishes: setDishes
            }}>
            {children
                //this indicates that all the child tags with MyProvider as Parent can access the global store  
            }
        </FoodContext.Provider>)
}

export default FoodProvider;