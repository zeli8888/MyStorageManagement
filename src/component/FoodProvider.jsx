import React, { useState, useEffect } from 'react';
import ingredientService from '../service/IngredientService'
import dishService from '../service/DishService'

export const FoodContext = React.createContext();  //exporting context object

const FoodProvider = ({ children }) => {
    const [allIngredients, setAllIngredients] = useState([]);
    const [allDishes, setAllDishes] = useState([]);

    useEffect(() => {
        ingredientService.getAllIngredients().then(response => { setAllIngredients(response.data) }).catch(error => { console.log(error) });
        dishService.getAllDishes().then(response => { setAllDishes(response.data) }).catch(error => { console.log(error) });
    }, []);

    return (
        <FoodContext.Provider value={
            {
                allIngredients: allIngredients,
                setAllIngredients: setAllIngredients,
                allDishes: allDishes,
                setAllDishes: setAllDishes
            }}>
            {children
                //this indicates that all the child tags with MyProvider as Parent can access the global store  
            }
        </FoodContext.Provider>)
}

export default FoodProvider;