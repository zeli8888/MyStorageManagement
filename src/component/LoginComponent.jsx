import * as React from 'react';
import { SignInPage } from '@toolpad/core/SignInPage';
import { Navigate, useNavigate, Link } from 'react-router';
import SessionContext from './UserProvider';
import LinearProgress from '@mui/material/LinearProgress';
import {
    signInWithGoogle,
    signInWithGithub,
    signInWithCredentials,
} from '../service/firebase/auth';
export default function LoginComponent() {
    const { session, setSession, loading } = React.useContext(SessionContext);
    const navigate = useNavigate();

    if (loading) {
        return <LinearProgress />;
    }

    if (session) {
        return <Navigate to="/" />;
    }

    return (
        <SignInPage
            sx={{
                '& .MuiContainer-root': {
                    height: 'auto !important',
                    minHeight: '100%'
                }
            }}
            providers={[
                { id: 'google', name: 'Google' },
                { id: 'github', name: 'GitHub' },
                { id: 'credentials', name: 'Email and Password' },
            ]}
            signIn={async (provider, formData, callbackUrl) => {
                let result;
                try {
                    if (provider.id === 'google') {
                        result = await signInWithGoogle();
                    }
                    if (provider.id === 'github') {
                        result = await signInWithGithub();
                    }
                    if (provider.id === 'credentials') {
                        const email = formData?.get('email');
                        const password = formData?.get('password');

                        if (!email || !password) {
                            return { error: 'Email and password are required' };
                        }

                        result = await signInWithCredentials(email, password);
                    }

                    if (result?.success && result?.user) {
                        // Convert Firebase user to Session format
                        const userSession = {
                            user: {
                                name: result.user.displayName || '',
                                email: result.user.email || '',
                                image: result.user.photoURL || '',
                            },
                        };
                        setSession(userSession);
                        navigate(callbackUrl || '/', { replace: true });
                        return {};
                    }
                    return { error: result?.error || 'Failed to sign in' };
                } catch (error) {
                    return {
                        error: error instanceof Error ? error.message : 'An error occurred',
                    };
                }
            }}
            slots={{
                signUpLink: () => <Link to="/register">Create Account</Link>,
                forgotPasswordLink: () => <Link to="/reset-password">Forgot Password?</Link>,
            }}
        />
    );
}