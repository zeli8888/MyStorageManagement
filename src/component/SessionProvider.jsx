import { createContext, useState, useEffect } from 'react';
import { onAuthStateChanged } from '../service/firebase/auth';

export const SessionContext = createContext();

const SessionProvider = ({ children }) => {
  const [session, setSession] = useState(null);
  const [loading, setLoading] = useState(true);
  const [rememberMe, setRememberMe] = useState(false);

  useEffect(() => {
    // Returns an `unsubscribe` function to be called during teardown
    const unsubscribe = onAuthStateChanged((user) => {
      if (user) {
        setSession({
          user: {
            name: user.displayName || '',
            email: user.email || '',
            image: user.photoURL || '',
          },
        });
      } else {
        setSession(null);
      }
      setLoading(false);
    });

    return () => unsubscribe();
  }, []);

  return (
    <SessionContext.Provider value={
      {
        session,
        loading,
        setSession,
        setLoading,
        rememberMe,
        setRememberMe
      }}>
      {children}
    </SessionContext.Provider>)
}

export default SessionProvider;
