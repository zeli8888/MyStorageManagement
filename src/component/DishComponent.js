import React, { useState, useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import Button from '@mui/material/Button';
import TextField from '@mui/material/TextField';
import Dialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import DialogContent from '@mui/material/DialogContent';
import DialogContentText from '@mui/material/DialogContentText';
import DialogTitle from '@mui/material/DialogTitle';
import DishService from '../service/DishService'
import { FoodContext } from './FoodProvider';
import PropTypes from 'prop-types';
import Box from '@mui/material/Box';
import Collapse from '@mui/material/Collapse';
import IconButton from '@mui/material/IconButton';
import Table from '@mui/material/Table';
import TableBody from '@mui/material/TableBody';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import TableHead from '@mui/material/TableHead';
import TableRow from '@mui/material/TableRow';
import Typography from '@mui/material/Typography';
import Paper from '@mui/material/Paper';
import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown';
import KeyboardArrowUpIcon from '@mui/icons-material/KeyboardArrowUp';
import MenuItem from '@mui/material/MenuItem';
import InputLabel from '@mui/material/InputLabel';
import ListItemText from '@mui/material/ListItemText';
import Select from '@mui/material/Select';
import Checkbox from '@mui/material/Checkbox';
import Grid from '@mui/material/Grid'
const DishComponent = function () {
    const navigate = useNavigate();
    const { ingredients, setIngredients, dishes, setDishes } = React.useContext(FoodContext);
    // const [dishes, setDishes] = useState([]);
    const [addingDish, setAddingDish] = useState(false);
    const [ingredientsForDish, setIngredientsForDish] = useState([]);

    useEffect(() => {
        refreshDishes();
    }, [navigate]);

    const refreshDishes = () => {
        DishService.getAllDishes().then(response => {
            setDishes(response.data);
        }).catch(error => {
            console.log(error);
        });
    }

    const updateDish = (data) => {
        DishService.updateDish(data.dishId, data).then(response => {
            refreshDishes();
        }).catch(error => {
            console.log(error);
        });
    }

    const addDish = (data) => {
        DishService.addDish(data).then(response => {
            setAddingDish(false);
            refreshDishes();
        }).catch(error => {
            console.log(error);
        });
    }

    const deleteDish = (dishId) => {
        DishService.deleteDish(dishId).then(response => {
            refreshDishes();
        }).catch(error => {
            console.log(error);
        });
    }

    function CreateDishRow(props) {
        const { dish } = props;
        const [open, setOpen] = React.useState(false);

        return (
            <React.Fragment>
                <TableRow sx={{ '& > *': { borderBottom: 'unset' } }}>
                    <TableCell>
                        <IconButton
                            aria-label="expand row"
                            size="small"
                            onClick={() => setOpen(!open)}
                        >
                            {open ? <KeyboardArrowUpIcon /> : <KeyboardArrowDownIcon />}
                        </IconButton>
                    </TableCell>
                    <TableCell component="th" scope="row">
                        {dish.dishName}
                    </TableCell>
                    <TableCell>{dish.dishDesc}</TableCell>
                </TableRow>
                <TableRow>
                    <TableCell style={{ paddingBottom: 0, paddingTop: 0 }} colSpan={3}>
                        <Collapse in={open} timeout="auto" unmountOnExit>
                            <Box sx={{ margin: 1 }}>
                                <Typography variant="h6" gutterBottom component="div">
                                    Recipe
                                </Typography>
                                <Table size="small" aria-label="purchases">
                                    <TableHead>
                                        <TableRow>
                                            <TableCell>Name</TableCell>
                                            <TableCell align="right">Amount</TableCell>
                                            <TableCell align="right">Storage</TableCell>
                                            <TableCell align="right">Cost</TableCell>
                                            <TableCell>Description</TableCell>
                                        </TableRow>
                                    </TableHead>
                                    <TableBody>
                                        {dish.dishIngredients.map((dishIngredient) => (
                                            <TableRow key={dishIngredient.ingredient.ingredientId}>
                                                <TableCell component="th" scope="row">
                                                    {dishIngredient.ingredient.ingredientName}
                                                </TableCell>
                                                <TableCell align="right">{dishIngredient.dishIngredientQuantity}</TableCell>
                                                <TableCell align="right">{dishIngredient.ingredient.ingredientStorage}</TableCell>
                                                <TableCell align="right">{dishIngredient.ingredient.ingredientCost}</TableCell>
                                                <TableCell>{dishIngredient.ingredient.ingredientDesc}</TableCell>
                                            </TableRow>
                                        ))}
                                    </TableBody>
                                </Table>
                            </Box>
                        </Collapse>
                    </TableCell>
                </TableRow>
            </React.Fragment>
        );
    }

    const addIngredientForDish = (event) => {
        const {
            target: { value },
        } = event;
        setIngredientsForDish(
            // On autofill we get a stringified value.
            typeof value === 'string' ? value.split(',') : value,
        );
    }

    CreateDishRow.propTypes = {
        dish: PropTypes.shape({
            dishId: PropTypes.number.isRequired,
            dishName: PropTypes.string.isRequired,
            dishDesc: PropTypes.string.isRequired,
            dishIngredients: PropTypes.arrayOf(
                PropTypes.shape({
                    dishIngredientQuantity: PropTypes.number.isRequired,
                    ingredient: PropTypes.shape({
                        ingredientId: PropTypes.number.isRequired,
                        ingredientName: PropTypes.string.isRequired,
                        ingredientStorage: PropTypes.number.isRequired,
                        ingredientCost: PropTypes.number.isRequired,
                        ingredientDesc: PropTypes.string.isRequired,
                    })
                }),
            ).isRequired,
        }).isRequired,
    };

    return (
        <div className="container">
            <TableContainer component={Paper}>
                <Table aria-label="collapsible table">
                    <TableHead>
                        <TableRow>
                            <TableCell />
                            <TableCell>Dish</TableCell>
                            <TableCell>Description</TableCell>
                        </TableRow>
                    </TableHead>
                    <TableBody>
                        {dishes.map((dish) => (
                            <CreateDishRow key={dish.dishId} dish={dish} />
                        ))}
                    </TableBody>
                </Table>
            </TableContainer>

            <Button variant="contained" color="success"
                onClick={() => {
                    setAddingDish(true);
                }}
            >
                New Dish
            </Button>

            <Dialog
                open={addingDish}
                onClose={() => { setAddingDish(false) }}
                maxWidth="xs"
                fullWidth={true}
                slotProps={{
                    paper: {
                        component: 'form',
                        onSubmit: (event) => {
                            event.preventDefault();
                            const formData = new FormData(event.currentTarget);
                            const formJson = Object.fromEntries(formData.entries());
                            addDish(formJson);
                        },
                    },
                }}
            >
                <DialogTitle>New Dish</DialogTitle>
                <DialogContent>

                    <TextField
                        autoFocus
                        required
                        margin="dense"
                        name="dishName"
                        label="Dish Name"
                        type="text"
                        fullWidth
                        variant="standard"
                    />
                    <TextField
                        margin="dense"
                        name="dishDesc"
                        label="Dish Description"
                        type="text"
                        fullWidth
                        variant="standard"
                    />
                    <Box mt={4} />
                    <InputLabel id="add-ingredient-for-dish">Recipe</InputLabel>
                    <Select
                        name="ingredientsForDish"
                        labelId="add-ingredient-for-dish"
                        multiple
                        value={ingredientsForDish}
                        onChange={addIngredientForDish}
                        renderValue={(selected) => selected.join(', ')}
                        MenuProps={{
                            PaperProps: { style: { maxHeight: '45%' } }
                        }}
                        style={{ width: '100%' }}
                    >
                        {ingredients.map((ingredient) => (
                            <MenuItem key={ingredient.ingredientId} value={ingredient.ingredientName}>
                                <Checkbox checked={ingredientsForDish.includes(ingredient.ingredientName)} />
                                <ListItemText primary={ingredient.ingredientName} />
                            </MenuItem>
                        ))}
                    </Select>
                    <Grid container spacing={2} >
                        {ingredientsForDish.map((ingredient) => {
                            return (
                                <TextField
                                    key={ingredient}
                                    name={ingredient}
                                    label={ingredient + " Amount"}
                                    value={1}
                                    variant="standard"
                                    color="success"
                                    margin="dense"
                                    type="number"
                                    required
                                />
                            );
                        })}
                    </Grid>
                </DialogContent>
                <DialogActions>
                    <Button onClick={() => { setAddingDish(false) }}>Cancel</Button>
                    <Button type="submit">Save</Button>
                </DialogActions>
            </Dialog>
        </div >
    )
}

export default DishComponent;