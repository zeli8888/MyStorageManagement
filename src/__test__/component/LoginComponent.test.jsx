import { describe, it, expect, vi, beforeEach } from 'vitest';
import { render, screen, cleanup, fireEvent } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import LoginComponent from '../../component/LoginComponent';
import { SessionContext } from '../../component/SessionProvider';
import {
    signInWithGoogle,
    signInWithGithub,
    signInWithMicrosoft,
    signInWithCredentials,
} from '../../service/firebase/auth';
import '@testing-library/jest-dom/vitest'
import { act } from 'react';

// Mock external dependencies
vi.mock('@toolpad/core/SignInPage', () => ({
    SignInPage: ({ providers, signIn, slots }) => (
        <div data-testid="signin-page">
            {providers.map((p) => {
                if (p.id === 'credentials') {
                    return (
                        <form
                            key={p.id}
                            data-testid="credentials-form"
                            onSubmit={(e) => {
                                e.preventDefault();
                                const formData = new FormData(e.target);
                                signIn(p, formData, '/');
                            }}
                        >
                            <input name="email" data-testid="email-input" />
                            <input name="password" data-testid="password-input" />
                            <button type="submit">{p.name}</button>
                        </form>
                    );
                }
                return (
                    <button
                        key={p.id}
                        data-testid={`provider-${p.id}`}
                        onClick={() => signIn(p, null, null)} // callback url null test
                    >
                        {p.name}
                    </button>
                );
            })}
            {slots.signUpLink()}
            {slots.forgotPasswordLink()}
            {slots.rememberMe()}
        </div>
    ),
}));

vi.mock('../../service/firebase/auth', () => ({
    signInWithGoogle: vi.fn(),
    signInWithGithub: vi.fn(),
    signInWithMicrosoft: vi.fn(),
    signInWithCredentials: vi.fn(),
}));

vi.mock('react-router', async (importOriginal) => ({
    ...(await importOriginal()),
    Navigate: ({ to }) => <div data-testid="navigate" data-to={to} />,
}));

const mockSessionContext = {
    session: null,
    setSession: vi.fn(),
    loading: false,
    rememberMe: false,
    setRememberMe: vi.fn(),
};

describe('LoginComponent', () => {
    beforeEach(() => {
        vi.clearAllMocks();
    });
    afterEach(cleanup)

    // Snapshot test for default render
    it('renders correctly and matches snapshot', () => {
        const { asFragment } = render(
            <MemoryRouter>
                <SessionContext.Provider value={mockSessionContext}>
                    <LoginComponent />
                </SessionContext.Provider>
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });

    // Test loading state
    it('shows loading spinner when loading', () => {
        render(
            <MemoryRouter>
                <SessionContext.Provider
                    value={{ ...mockSessionContext, loading: true }}
                >
                    <LoginComponent />
                </SessionContext.Provider>
            </MemoryRouter>
        );
        expect(screen.getByRole('progressbar')).toBeInTheDocument();
    });

    // Test redirection when session exists
    it('redirects to home when session exists', () => {
        render(
            <MemoryRouter>
                <SessionContext.Provider
                    value={{ ...mockSessionContext, session: { user: {} } }}
                >
                    <LoginComponent />
                </SessionContext.Provider>
            </MemoryRouter>
        );
        expect(screen.getByTestId('navigate')).toBeInTheDocument();
    });

    // Test provider sign-in buttons
    it('has all provider buttons', () => {
        render(
            <MemoryRouter>
                <SessionContext.Provider value={mockSessionContext}>
                    <LoginComponent />
                </SessionContext.Provider>
            </MemoryRouter>
        );
        expect(screen.getByText('Google')).toBeInTheDocument();
        expect(screen.getByText('GitHub')).toBeInTheDocument();
        expect(screen.getByText('Microsoft')).toBeInTheDocument();
        expect(screen.getByText('Email and Password')).toBeInTheDocument();
    });

    // Test helper links
    it('contains sign up and password reset links', () => {
        render(
            <MemoryRouter>
                <SessionContext.Provider value={mockSessionContext}>
                    <LoginComponent />
                </SessionContext.Provider>
            </MemoryRouter>
        );
        expect(screen.getByText('Create Account')).toBeInTheDocument();
        expect(screen.getByText('Forgot Password?')).toBeInTheDocument();
    });

    // Test successful Google sign-in
    it('handles Google sign-in successfully', async () => {
        signInWithGoogle.mockResolvedValue({
            success: true,
            user: {
                displayName: 'Test User',
                email: 'test@example.com',
                photoURL: 'photo.jpg'
            }
        });

        render(
            <MemoryRouter>
                <SessionContext.Provider value={mockSessionContext}>
                    <LoginComponent />
                </SessionContext.Provider>
            </MemoryRouter>
        );

        await act(async () => {
            fireEvent.click(screen.getByText('Google'));
        })

        await vi.waitFor(() => {
            expect(signInWithGoogle).toHaveBeenCalledWith(false);
            expect(mockSessionContext.setSession).toHaveBeenCalledWith({
                user: {
                    name: 'Test User',
                    email: 'test@example.com',
                    image: 'photo.jpg'
                }
            });
        });
    });

    // Test successful Github sign-in
    it('handles Github sign-in successfully', async () => {
        signInWithGithub.mockResolvedValue({
            success: true,
            user: {
                displayName: null,
                email: null,
                photoURL: null
            }
        });

        render(
            <MemoryRouter>
                <SessionContext.Provider value={mockSessionContext}>
                    <LoginComponent />
                </SessionContext.Provider>
            </MemoryRouter>
        );

        await act(async () => {
            fireEvent.click(screen.getByText('GitHub'));
        })

        await vi.waitFor(() => {
            expect(signInWithGithub).toHaveBeenCalledWith(false);
            expect(mockSessionContext.setSession).toHaveBeenCalledWith({
                user: {
                    name: '',
                    email: '',
                    image: ''
                }
            });
        });
    });

    // Test failed GitHub sign-in with error
    it('handles GitHub sign-in failure', async () => {
        signInWithGithub.mockRejectedValue(new Error('Authentication failed'));

        render(
            <MemoryRouter>
                <SessionContext.Provider value={mockSessionContext}>
                    <LoginComponent />
                </SessionContext.Provider>
            </MemoryRouter>
        );

        await act(async () => {
            fireEvent.click(screen.getByText('GitHub'));
        })

        await vi.waitFor(() => {
            expect(signInWithGithub).toHaveBeenCalledWith(false);
            expect(mockSessionContext.setSession).not.toHaveBeenCalled();
        });
    });

    // Test failed GitHub sign-in with standard error message
    it('handles GitHub sign-in failure with standard error message', async () => {
        signInWithGithub.mockResolvedValue({
            success: false,
            error: 'test error message'
        });

        render(
            <MemoryRouter>
                <SessionContext.Provider value={mockSessionContext}>
                    <LoginComponent />
                </SessionContext.Provider>
            </MemoryRouter>
        );

        await act(async () => {
            fireEvent.click(screen.getByText('GitHub'));
        })

        await vi.waitFor(() => {
            expect(signInWithGithub).toHaveBeenCalledWith(false);
            expect(mockSessionContext.setSession).not.toHaveBeenCalled();
        });
    });

    // Test successful email/password sign-in
    it('handles email/password sign-in successfully', async () => {
        signInWithCredentials.mockResolvedValue({
            success: true,
            user: {
                displayName: 'Test User',
                email: 'test@example.com',
                photoURL: 'photo.jpg'
            }
        });

        render(
            <MemoryRouter>
                <SessionContext.Provider value={mockSessionContext}>
                    <LoginComponent />
                </SessionContext.Provider>
            </MemoryRouter>
        );

        await act(async () => {
            fireEvent.change(screen.getByTestId('email-input'), {
                target: { value: 'test@example.com' }
            });
            fireEvent.change(screen.getByTestId('password-input'), {
                target: { value: 'password123' }
            });
            fireEvent.submit(screen.getByTestId('credentials-form'));
        })

        await vi.waitFor(() => {
            expect(signInWithCredentials).toHaveBeenCalledWith(
                false,
                'test@example.com',
                'password123'
            );
            expect(mockSessionContext.setSession).toHaveBeenCalledWith({
                user: {
                    name: 'Test User',
                    email: 'test@example.com',
                    image: 'photo.jpg'
                }
            });
        });
    });

    it('shows error when email/password is missing', async () => {

        render(
            <MemoryRouter>
                <SessionContext.Provider value={mockSessionContext}>
                    <LoginComponent />
                </SessionContext.Provider>
            </MemoryRouter>
        );

        act(() => {
            fireEvent.submit(screen.getByTestId('credentials-form'));
        })

        await vi.waitFor(() => {
            expect(signInWithCredentials).not.toHaveBeenCalled();
        });
    });

    // Test remember me functionality
    it('passes rememberMe preference to auth providers', async () => {
        const customContext = {
            ...mockSessionContext,
            rememberMe: true
        };

        render(
            <MemoryRouter>
                <SessionContext.Provider value={customContext}>
                    <LoginComponent />
                </SessionContext.Provider>
            </MemoryRouter>
        );

        act(() => {
            fireEvent.click(screen.getByText('Microsoft'));
        })

        await vi.waitFor(() => {
            expect(signInWithMicrosoft).toHaveBeenCalledWith(true);
        });
    });

});