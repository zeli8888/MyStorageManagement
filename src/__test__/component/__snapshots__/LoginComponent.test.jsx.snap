// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoginComponent > renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="signin-page"
  >
    <button
      data-testid="provider-google"
    >
      Google
    </button>
    <button
      data-testid="provider-github"
    >
      GitHub
    </button>
    <button
      data-testid="provider-microsoft"
    >
      Microsoft
    </button>
    <form
      data-testid="credentials-form"
    >
      <input
        data-testid="email-input"
        name="email"
      />
      <input
        data-testid="password-input"
        name="password"
      />
      <button
        type="submit"
      >
        Email and Password
      </button>
    </form>
    <a
      data-discover="true"
      href="/register"
    >
      Create Account
    </a>
    <a
      data-discover="true"
      href="/reset-password"
    >
      Forgot Password?
    </a>
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-14r958l-MuiFormControlLabel-root"
    >
      <span
        class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeSmall PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeSmall MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeSmall css-reoaak-MuiButtonBase-root-MuiSwitchBase-root-MuiCheckbox-root"
      >
        <input
          class="PrivateSwitchBase-input css-12xagqm-MuiSwitchBase-root"
          data-indeterminate="false"
          name="remember"
          type="checkbox"
          value="true"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
          data-testid="CheckBoxOutlineBlankIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
      </span>
      <span
        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-1c26xtj-MuiTypography-root"
      >
        Remember me
      </span>
    </label>
  </div>
</DocumentFragment>
`;
